### PR TITLE
Fix exception and searching for maintainers page

### LINKED ISSUE #63: Broken maintainer's page when invalid maintainer handle
We need to rethink a little around the maintainer's page. I was of the opinion that when clicked on a maintainer's name in the port-summary page, it'll simply go to another page but instead, it tries to load the content over it. (Browser's backward and forward buttons don't work with the maintainer's page, to verify). We really don't need any in-place modification of content like this.

Also, passing a wrong maintainer's handle breaks the app on accessing `all_ports` variable, e.g. http://\<ec2\>/maintainer/github/`umesh`.  The correct URL was http://\<ec2\>/maintainer/github/`umeshksingla`. I don't think the query gets fired.

### PR DESCRIPTION
**Pull request by arjunsalyan**
Closes: https://github.com/macports/macports-webapp/issues/63

### CODE REVIEW

File: app/ports/views.py [hunk 1]
```diff
@@ -468,9 +472,9 @@ def search_ports_in_category(request):
 # Respond to ajax calls for searching within a maintainer
 def search_ports_in_maintainer(request):
     query = request.GET.get('name', '')
-    name = request.GET.get('maintainers__name')
+    name = request.GET.get('maintainers__name', '')
     github = request.GET.get('maintainers__github')
-    if name is None:
+    if name is None or name == '':
```
Comment (umeshksingla) [inline app/ports/views.py:477]: Will `name` ever be `None`?

File: app/ports/views.py [hunk 2]
```diff
@@ -403,6 +405,8 @@ def maintainer_detail_github(request, github_handle):
 
 def maintainer_detail_email(request, name, domain):
     maintainers = Maintainer.objects.filter(name=name, domain=domain)
+    if not maintainers.count() > 0:
```
Comment (umeshksingla) [inline app/ports/views.py]: I think we can write it in a more straightforward way which to me is more readable: `if maintainers.count() == 0`. What do you think?

Comment (arjunsalyan) [reply app/ports/views.py]: Seems fine to me.

File: app/ports/views.py (hunk 1 above)
Comment (arjunsalyan) [reply app/ports/views.py:477]: Probably never, but I thought it won't hurt to have it. Shall I remove this?

Comment (umeshksingla) [reply app/ports/views.py:477]: It should be fine. Though I hope we don't incur a case where both `name` and `github` are `''`.

**Review by umeshksingla — APPROVED** [review]

File: app/ports/views.py (hunk 1 above)
Comment (jmroot) [reply app/ports/views.py:477]: That would be a mistake in the portfile, but you should make sure you don't break if it does happen.

**Pull request merged**